{{cookiecutter.project_slug}}/fabfile.py

```diff
@@ -1,144 +1,234 @@
+        if isinstance(value, bool):
+            value = int(value)
+
+        envfile_path = os.path.join(self.envdirpath, name)
+
+        will_write = force
+        try:
+            # Test that it does exist
+            self.run_in_wwwroot("test -r {}".format(envfile_path), hide=True)
+        except UnexpectedExit:
+            will_write = True
+
+        if will_write:
+            self.put(StringIO("{}\n".format(value)), envfile_path)
+
+    def dump_db(self, destination):
+        """
+        Dump the database to the given directory and return the path to the file created.
+        This creates a gzipped SQL file.
+        """
+        with self.cd(self.wwwroot):
+            db_credentials = self.run("cat envdir/DATABASE_URL", hide=True).stdout.strip()
+        db_credentials_dict = dj_database_url.parse(db_credentials)
+
+        if not is_supported_db_engine(db_credentials_dict["ENGINE"]):
+            raise NotImplementedError(
+                "The dump_db task doesn't support the remote database engine"
+            )
 
+        outfile = os.path.join(
+            destination, datetime.now().strftime("%Y-%m-%d_%H%M%S.sql.gz")
+        )
 
-def run_python(args):
-    """
-    Run the python command in the remote virtualenv.
-    """
-    return run_in_virtualenv("python", args)
+        self.run(
+            "pg_dump -O -x -h {host} -U {user} {db}|gzip > {outfile}".format(
+                host=db_credentials_dict["HOST"],
+                user=db_credentials_dict["USER"],
+                db=db_credentials_dict["NAME"],
+                outfile=outfile,
+            ),
+            env={"PGPASSWORD": db_credentials_dict["PASSWORD"].replace("$", "\$")},
+        )
 
+        return outfile
 
-def install_requirements():
-    """
-    Install the requirements from the base.txt file to the remote virtualenv.
-    """
-    with cd(get_project_root()):
-        run_pip("install -r requirements/base.txt")
+    def create_structure(self):
+        """
+        Create the basic directory structure on the remote server.
+        """
+        self.run("mkdir -p %s" % self.wwwroot)
 
+        with self.cd(self.project_root):
+            self.run("mkdir -p static backups media")
+            self.run("python3 -m venv venv")
 
-def migrate_database():
-    with cd(get_project_root()):
-        run_python("manage.py migrate")
+    def clean_old_database_backups(self, nb_backups_to_keep):
+        """
+        Remove old database backups from the system and keep `nb_backups_to_keep`.
+        """
+        backups = self.ls(self.backups_root)
+        backups = sorted(backups, reverse=True)
 
+        if len(backups) > nb_backups_to_keep:
+            backups_to_delete = backups[nb_backups_to_keep:]
 
-def collect_static():
-    """
-    Collect static files to the STATIC_ROOT directory.
-    """
-    with cd(get_project_root()):
-        run_python("manage.py collectstatic --noinput")
+            for backup_to_delete in backups_to_delete:
```

Comment:
I think we could issue only one `rm` command with all the backups paths to delete, no?